python/lsst/validate/base/specset.py

```diff
@@ -0,0 +1,562 @@
+# See COPYRIGHT file at the top of the source tree.
+from __future__ import print_function, division
+
+from collections import OrderedDict
+import os
+import re
+
+from lsst.utils import getPackageDir
+
+from .spec.base import Specification
+from .spec.threshold import ThresholdSpecification
+from .naming import Name
+from .errors import SpecificationResolutionError
+from .yamlutils import merge_documents, load_all_ordered_yaml
+
+__all__ = ['SpecificationSet']
+
+
+# Pattern for SpecificationPartial names
+# package:path#name
+PARTIAL_PATTERN = re.compile('^(?:(?P<package>\S+):)'
+                             '?(?P<path>\S+)?#(?P<name>\S+)$')
+
+
+class SpecificationSet():
+    """A collection of Specifications.
+
+    Parameters
+    ----------
+    specifications : `list` or `tuple` of `Specification` instances
+        A sequence of `Specification` instances.
+    partials : `list` or `tuple` of `SpecificationPartial` instances
+        A sequence of `SpecificationPartial` instances. These partials
+        can be used as bases for specification definitions.
+    """
+
+    def __init__(self, specifications=None, partials=None):
+        # Specifications, keyed by Name (a specification name)
+        self._specs = {}
+
+        # SpecificationPartial instances, keyed by the fully-qualified
+        # name: ``package_name:yaml_id#name``.
+        self._partials = {}
+
+        if specifications is not None:
+            for spec in specifications:
+                if not isinstance(spec, Specification):
+                    message = '{0!r} must be a Specification type'
+                    raise TypeError(message.format(spec))
+
+                self._specs[spec.name] = spec
+
+        if partials is not None:
+            for partial in partials:
+                if not isinstance(partial, SpecificationPartial):
+                    message = '{0!r} must be a SpecificationPartial type'
+                    raise TypeError(message.format(partial))
+
+                self._partials[partial.name] = partial
+
+    @classmethod
+    def load_eups_package(cls, package_name='validate_metrics'):
```

Comment:
We've tried really hard to remove eups mentions from the codebase where we don't explicitly have an eups dependency. This function is only using `getPackageDir` so please rename this function.